### DISCUSSION TITLE
Is there a way to force grouped updates?

### DISCUSSION THREAD

**Discussion by lauraseidler** [renovatebot/renovate#9457] [Request Help] [answered]
I am running the following config:

```
  "packageRules": [
    {
      "groupName": "Cypress",
      "managers": ["dockerfile", "npm"],
      "packagePatterns": ["^cypress/included$", "^cypress$"],
      "additionalBranchPrefix": ""
    }
  ]
  ```
  
This groups updates to the `cypress/included` docker tag as well as the `cypress` npm package - which generally works fine.
However, usually the npm package is published about a day before the docker image, which leads to the following process:

1. Renovate opens a PR for the npm package
2. The build fails since the versions don't match
3. As soon as the docker image update is available, renovate adapts the PR
4. Now the build passes, and the PR can be merged

Is there a way to force this grouping and only open a PR if all grouped packages have updates? 

I've also thought about leveraging `stabilityDays`, but I'm unsure whether this would work to set only for this particular group, as there are other unrelated package rules in my configuration.

Thanks!

**Comment by viceice** [comment, marked as answer]
`stabilityDays=1` added to your rule should defer the updates by one day and do the trick

**Reply by rarkins** [reply]
Combined with prCreation=pending?

**Reply by lauraseidler** [reply]
Hi @rarkins `pending` doesn't seem to be a valid option? https://docs.renovatebot.com/configuration-options/#prcreation
Did you mean `non-pending`? 

If I understand correctly, this would create the PR once the branch build succeeds. We currently don't build branches without open PRs - so I guess this wouldn't work for us? 

Indeed with just `stabilityDays`, it will open the PR immediately, and just add an extra status check that only passes after the given amount of days.

**Reply by rarkins** [reply]
Yes, I meant not-pending. Does your CI have an option to build renovate/* branches even if no PR exists?

**Reply by lauraseidler** [reply]
We could enable this, but I would like to avoid it - we would have to enable it on every project using cypress manually (our renovate config is shared), and it would consume resources for builds we already know will fail.

**Reply by rarkins** [reply]
If you add `stabilityDays=1` to your rule like @viceice suggested, you'll get the PR created right away with 1 package updated, but it will also have a pending `renovate/stability-days` check. I don't think it alone solves your problem.

If you added `prCreation=not-pending` but none of your own tests run, then I think you'd see that:
- The commit has only one status check.. Renovate's.. and it's pending
- After 1 day it will switch from pending to passed, and then the PR is created soon after
- Right after the PR is created, your regular CI will kick off, adding new checks

The above should hopefully all work well as long as... the Docker image is created within that 24 hours. It's not exactly what you asked for, e.g.
- If the image is created after 12 hours then you don't get the PR until at least 24-25 hours
- If the image is created after 26 hours then the PR would still get created anyway

But it might be close enough?

**Reply by lauraseidler** [reply]
I think that would definitely be close enough - delaying updates for a day or two is perfectly fine for us, as long as we don't get unnecessarily pinged for failing builds. I will give it a spin and see if it works as you say. Thanks very much for the detailed response!

**Reply by lauraseidler** [reply]
so it took a while until the next Cypress update happened, but I can now confirm this works fine - thanks!
